subsys/bluetooth/mesh/adv.c

```diff
@@ -41,8 +41,15 @@ const uint8_t bt_mesh_adv_type[BT_MESH_ADV_TYPES] = {
 
 K_FIFO_DEFINE(bt_mesh_adv_queue);
 
+static void adv_buf_destroy(struct net_buf *buf)
+{
+	bt_mesh_adv_send_end(0, BT_MESH_ADV(buf));
```

Comment:
Im a little bit worried about this, since the buffer being freed doesn't automatically mean that this is the moment when sending stopped. There could be parts of the stack keeping its own references to the buffer (in fact, doesn't the transport layer do this for segmented messages?) in which case the "send_end" callback would be called much too late. Or am I misunderstanding something?

Reply:
> in fact, doesn't the transport layer do this for segmented messages?

The problem of segment messages occupying the adv buffer has been resolved in PR(https://github.com/zephyrproject-rtos/zephyr/pull/23398).

Reply:
@trond-snekvik Any think about?